src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.MemoryMappedInfo.cs

```diff
@@ -143,38 +174,25 @@ private void Dispose(bool disposing)
             {
                 if (disposing)
                 {
-                    lock (_memoryMappedFile)
-                    {
-                        if (_accessor != null)
-                        {
-                            // (see remarks on accessor for relation between _accessor and the streams)
-                            _accessor.Dispose();
-                            _accessor = null;
-                        }
-                    }
-
-                    // (see remarks on accessor for relation between _memoryMappedFile and the views/streams)
+                    // See remarks on field for relation between _memoryMappedFile and the views/streams. There is no
+                    // need to write _weakAccessor here since the types involved adhere to their contracts even in
```

Comment:
Might be worth replacing this sentence with "There is no need to write _weakAccessor, since it will return null for any further reads". Basically, state which contract you're relying on, so when somebody changes the contract and doesn't update this it's more clear what happened.